docs/design/proposals/split-repo/split-repo.md

```diff
@@ -0,0 +1,103 @@
+images) are built and pushed to the [ECR public ACK controller image][ecr]
+registry with an image tag of `$SERVICE-$RELEASE_VERSION` -- e.g. `s3-v0.0.2`
+
+[ecr]: https://gallery.ecr.aws/aws-controllers-k8s/controller
+
+### Problem #1: Release series inflexibility
+
+This monorepo tightly couples the ACK common runtime code, code
+generator/templates and the individual service controller code, making the
+release process inflexible and cumbersome. The Elasticache team cannot have
+separate release series -- say, a `stable` and a `preview` release series --
+because releases are Git tags, and Git tags are specific to one source
+repository. If we had a `stable` and `preview` Git branch, all service
+controllers would need to follow the same release cadence and series naming. In
+addition, numeric releases such as `v0.2.0` refer to a specific
+content-addressable SHA1 Git commit, and with all service controllers in the
+same source repository, this means that numeric Git tags cannot applied to two
+different service controllers to represent an incremental change in one
+specific service controller's codebase.
+
+### Problem #2: Common runtime import inflexibility
+
+Furthermore, due to the code generator and ACK common runtime code being in the
+same repo as the service controllers that `import` this code, it's not easy to
+depend on a specific version of that common runtime from within the service
+controller code itself. Individual service controller package directories would
+need their own `go.mod` files and managing this would be a pain and wouldn't
+solve Problem #1 anyway.
+
+## Solution
+
+We propose splitting the monorepo into a set of source code repositories that
+can be separately Git-tagged and have their releases managed according to the
+preferences of the team that owns that component or individual ACK service
+controller.
+
+Specifically, we propose the following changes:
+
+1) All code for generating code will move into a new
+   `github.com/aws/aws-controllers-k8s-generate` source repository.
+
+   This includes all content in the existing `/cmd/ack-generate`, `/templates`,
+   `/pkg/generate` and `/pkg/model` directories of the existing
+   `github.com/aws/aws-controllers-k8s` source repository.
+
+2) The ACK common runtime and core ACK types will remain in the
+   `github.com/aws/aws-controllers-k8s` source repository.
+
+   The `/test` directory will contain only common testing framework code (both
+   Python and Bash-based for the immediate future).
+
+3) Each ACK service controller will move into a new
+   `github.com/aws/aws-controllers-k8s-$SERVICE` source repository.
+
+   There will be a single Git branch `main` for these individual service
+   controller source repositories. Teams that own individual service
+   controllers will be free to create additional Git branches for release
+   series -- e.g. `stable` or `preview` -- as they see fit.
```

Comment:
Once again, for any contributors external to the team, each service maintaining their own sets of standards could be confusing.

Reply:
https://github.com/aws/aws-controllers-k8s/blob/main/docs/contents/releases.md mentions use of following release statuses:

    PROPOSED
    PLANNED
    BUILD
    DEVELOPER PREVIEW
    BETA
    GENERALLY AVAILABLE (GA)

The proposed `main`, `stable` branches map to which of the release status?
Should there be branch that correspond to each of these release statuses?